File: Medium/2849. Determine if a Cell Is Reachable at a Given Time/2849. Determine if a Cell Is Reachable at a Given Time.py
class Solution:
    def isReachableAtTime(self, sx: int, sy: int, fx: int, fy: int, t: int) -> bool:
        dx = abs(sx - fx)
        dy = abs(sy - fy)

        if dx == 0 and dy == 0:
            return t != 1
        else:
            return max(dx, dy) <= t

File: Medium/2849. Determine if a Cell Is Reachable at a Given Time/2849. Determine if a Cell Is Reachable at a Given Time.rs
impl Solution {
    pub fn is_reachable_at_time(sx: i32, sy: i32, fx: i32, fy: i32, t: i32) -> bool {
        let dx = (sx - fx).abs();
        let dy = (sy - fy).abs();
        if dx == 0 && dy == 0 {
            t != 1
        } else {
            dx.max(dy) <= t
        }
    }
}
